File: mobile/src/api/notifications.api.js
import { http } from './http';

export const fetchNotifications = () => http.get('/notifications').then((r) => r.data);
export const fetchUnreadCount = () => http.get('/notifications/unread-count').then((r) => r.data);
export const markNotificationRead = (id) => http.post(`/notifications/${id}/read`).then((r) => r.data);
export const markAllNotificationsRead = () => http.post('/notifications/read-all').then((r) => r.data);

File: mobile/lib/features/notifications/data/notifications_api.dart
import 'package:dio/dio.dart';
import 'package:flutter_riverpod/flutter_riverpod.dart';

import '../../../core/api_client.dart';
import '../../../models/notification.dart';

/// Mirrors the web app's `src/api/notifications.api.js`. The backend caps
/// listNotifications at a fixed 50 rows (notification.controller.js) with no
/// before/limit query params, so there's no pagination to mirror here.
class NotificationsApi {
  final Dio _dio;

  NotificationsApi(this._dio);

  Future<List<AppNotification>> fetchNotifications() async {
    final res = await _dio.get<List<dynamic>>('/notifications');
    return res.data!.map((e) => AppNotification.fromJson(e as Map<String, dynamic>)).toList();
  }

  Future<int> fetchUnreadCount() async {
    final res = await _dio.get<Map<String, dynamic>>('/notifications/unread-count');
    return res.data!['count'] as int;
  }

  Future<void> markRead(String id) => _dio.post<void>('/notifications/$id/read');

  Future<void> markAllRead() => _dio.post<void>('/notifications/read-all');
}

final notificationsApiProvider = Provider<NotificationsApi>((ref) => NotificationsApi(ref.read(dioProvider)));
